### PR TITLE
chore: adds test to ToolDocumentCollatorFactory

### PR DESCRIPTION
**Pull request by codingdiaz**
## Hey, I just made a Pull Request!

     
Adds a unit test for the ToolDocumentCollatorFactory as part of the explore search collator.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/search-backend-module-explore/src/collators/ToolDocumentCollatorFactory.test.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+    it('runs against mock tools', async () => {
+      const pipeline = TestPipeline.fromCollator(collator);
+      const { documents } = await pipeline.execute();
+      expect(mockDiscoveryApi.getBaseUrl).toHaveBeenCalledWith('explore');
+      expect(documents).toHaveLength(mockTools.tools.length);
+    });
+
+    it('non-authenticated backend', async () => {
+      // Provide an alternate location template.
```
Comment (emmaindal) [inline plugins/search-backend-module-explore/src/collators/ToolDocumentCollatorFactory.test.ts:108]: was this comment meant to be there? I'm not seeing a custom location template 🤔

**Review by emmaindal — APPROVED** [review]
what a wonderful contribution, thanks a lot 💯 just a small comment from my side but will approve to unblock!

File: plugins/search-backend-module-explore/src/collators/ToolDocumentCollatorFactory.test.ts (hunk 1 above)
Comment (codingdiaz) [reply plugins/search-backend-module-explore/src/collators/ToolDocumentCollatorFactory.test.ts]: oops, that was not supposed to be there. I just removed it. Thanks for the catch @emmaindal!

**Review by benjdlambert — APPROVED** [review]
Nice thanks! :pray:

**Pull request merged**